File: java/0167-two-sum-ii-input-array-is-sorted.java
class Solution {

    public int[] twoSum(int[] numbers, int target) {
        int a_pointer = 0;
        int b_pointer = numbers.length - 1;
        int num_a, num_b;

        while (a_pointer < b_pointer) {
            num_a = numbers[a_pointer];
            num_b = numbers[b_pointer];

            if (num_a + num_b == target) break;

            if (num_a + num_b < target) {
                a_pointer++;
                continue;
            }

            b_pointer--;
        }

        return new int[] { a_pointer + 1, b_pointer + 1 };
    }
}

File: scala/0167-two-sum-ii-input-array-is-sorted.scala
object Solution {
  def twoSum(numbers: Array[Int], target: Int): Array[Int] = {
    def loop(p1: Int, p2: Int): Array[Int] = {
      if(numbers(p1) + numbers(p2) == target) Array(p1 + 1, p2 + 1)
      else if(numbers(p1) + numbers(p2) > target) loop(p1, p2 - 1)
      else loop(p1 + 1, p2)
    }
    loop(0, numbers.length - 1)
  }
}
